src/rules/html-self-closing.ts

```diff
@@ -27,32 +27,65 @@ export default createRule("html-self-closing", {
     },
     schema: [
       {
-        type: "object",
-        properties: {
-          void: {
-            enum: ["never", "always", "ignore"],
+        oneOf: [
+          {
+            properties: {
+              void: {
+                enum: ["never", "always", "ignore"],
+              },
+              normal: {
+                enum: ["never", "always", "ignore"],
+              },
+              component: {
+                enum: ["never", "always", "ignore"],
+              },
+              svelte: {
+                enum: ["never", "always", "ignore"],
+              },
+            },
+            additionalProperties: false,
           },
-          normal: {
-            enum: ["never", "always", "ignore"],
+          {
+            enum: ["all", "html", "none"],
           },
-          component: {
-            enum: ["never", "always", "ignore"],
-          },
-          svelte: {
-            enum: ["never", "always", "ignore"],
-          },
-        },
-        additionalProperties: false,
+        ],
       },
     ],
   },
   create(ctx) {
-    const options = {
+    let options = {
       void: "always",
       normal: "always",
       component: "always",
       svelte: "always",
-      ...ctx.options?.[0],
+    }
+
+    if (typeof ctx.options?.[0] === "object") {
+      options = {
+        ...options,
+        ...(ctx.options?.[0] ?? {}),
+      }
+    } else {
+      switch (ctx.options?.[0] ?? "") {
```

Comment:
```suggestion
      const option = ctx.options?.[0]
      switch (option) {
```

So that `option` can be reused in the default case as I described above.